### PR TITLE
Hold a rw lock on the cache before Get() for it can update the cache

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: kv/leader_cache.go [hunk 1]
```diff
@@ -48,9 +48,9 @@ func newLeaderCache(size int) *leaderCache {
 // Lookup consults the cache for the replica cached as the leader of
 // the given Raft consensus group.
 func (lc *leaderCache) Lookup(group proto.RaftID) *proto.Replica {
-	lc.mu.RLock()
+	lc.mu.Lock()
```
Comment (tschottdorf) [inline kv/leader_cache.go]: Change the type from `mu` from `RWMutex` to `Mutex`.

**Comment by tschottdorf** [comment]
LGTM

File: kv/leader_cache.go (hunk 1 above)
Comment (tschottdorf) [inline kv/leader_cache.go]: @vivekmenezes Did you not see this?

Comment (vivekmenezes) [inline kv/leader_cache.go]: I didn't see your comment. Will make that change.

**Pull request merged**